src/crimson/os/seastore/extent_placement_manager.h

```diff
@@ -122,6 +130,16 @@ class RandomBlockOolWriter : public ExtentOolWriter {
     return rb_cleaner->alloc_paddr(length);
   }
 
+  bool can_inplace_rewrite(Transaction& t,
+    CachedExtentRef extent) final {
+    if (t.get_src() == transaction_type_t::TRIM_DIRTY &&
+	get_extent_category(extent->get_type()) == data_category_t::DATA &&
+	extent->is_dirty()) {
+      return true;
+    }
+    return false;
```

Comment:
`can_inplace_rewrite()` is something like a query that is possible to be called from anywhere, like `epm->can_inplace_rewrite() `in `rewrite_extent()` or `can_inplace_rewrite() in do_write()` from `extent_placement_manager`, so that we need to return false if the extent is not a target without `assert()`. For example,
```

    if (t.get_src() != transaction_type_t::TRIM_DIRTY ||
        !extent->is_dirty()) {
      return false;
    }
    return crimson::os::seastore::can_inplace_rewrite(extent->get_type());
```

Without this code, I encountered several `assert` during tests.